flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -198,10 +211,47 @@ else if (message instanceof SubmitJobAndWait) {
 					decorateMessage(new Status.Failure(new Exception(msg))), ActorRef.noSender());
 			}
 		}
+		else if (message instanceof AttachToJobAndWait) {
```

Comment:
At the moment the `JobClientActor` can be used to attach to a job and submit a new job at the same time. Then depending on the order in which the jobs terminate either the submitted job's or the attached job's result is reported back but the other future will never be completed. It might make sense to guard against this or to have two different JobClientActors based on the same base class.